addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2688,10 +2688,6 @@ const SnippetOptionWidget = Widget.extend({
             }
         }
 
-        if (previewMode === false) {
-            this.wysiwyg.snippetsMenu.activateLastSnippetBlock();
-        }
-
```

Comment:
@Goaman I think you're the one who had added these lines. Can you please confirm it's now ok to remove them, or tell me what I might break by removing them again?

Reply:
@Goaman any input?

Reply:
`await this[methodName](previewMode, widgetValue, params);`
or 
`return this[methodName].call(proxy, previewMode, widgetValue, params);`
might call `updateChangesInWysiwyg`, which will invalidate reference for all the descendants of the `option.$target`.

If you remove that, the last "snippet block" will not be reactivated, therefore the `SnippetEditor`s will not be updated thus the related `$target` of their `SnippetOption`s will be wrong.

That should generate a traceback if you try to change an option of a `SnippetEditor` that is a children of the one that triggered the `_select`.

Why did you wanted to remove those lines?

Reply:
Because it does exactly what it says it does and that's not the spec :shrug:
Namely, it reopens the options of the last opened snippet at moments we don't expect it to.
Specifically, for instance, click in a snippet to open its options, then go to the options tab and change the theme. It will reopen the style tab which displays the options of the last opened snippet. We don't want that: it's confusing, it's different from 14.0 and it was reported to us as a bug.

Reply:
It's not meant to be a feature. It's a technical workaround.

I see multiples possible fix that you can do:
- move `this.wysiwyg.snippetsMenu.activateLastSnippetBlock()` within `updateChangesInWysiwyg`
- or split the logic that systematically change the tab when activating the `activateLastSnippetBlock` and maybe rename the function in something like `updateSnippetEditorReferences`

In any case, check that you are not introducing new bugs by testing different combination of option (especially testing an option of different `SnippetEditor`s)

Reply:
To sum up our audio discussion, this will actually be fixed with @Goaman 's branch so I'm removing this commit from here and moving the issue to his branch.